### ISSUE TITLE
[CS8602] Dereference of a possibly null reference when using switch with tuple containg nullable members and using not null pattern

### ISSUE THREAD

**Issue by Berreek** [dotnet/roslyn#49353]
**Version Used**: 
.NET 5 SDK 5.0.100

**Steps to Reproduce**:
```csharp
#nullable enable
using System;
using System.Collections.Generic;
using System.Linq;

var somethings = new List<Foo>();
var something1 = somethings.ElementAtOrDefault(1);
var something2 = somethings.ElementAtOrDefault(2);
switch (something1, something2)
{
    case (not null, not null):
        Console.WriteLine(something1.Bar);
        break;
}

public record Foo(string Bar); 
```
**Expected Behavior**:
Do not return [CS8602] Dereference of a possibly null reference since it was checked that the values are not null.

**Actual Behavior**:
[CS8602] Dereference of a possibly null reference

**Comment by RikkiGibson** [comment]
I think we are seeing this reported a lot, should prioritize for 16.9.

**Comment by jcouv** [comment]
Verified this is fixed by https://github.com/dotnet/roslyn/pull/51001 (merged for 16.10)
I'll close the issue. Thanks
